src/containers/Dev/Snapshots.tsx

```diff
@@ -0,0 +1,266 @@
+                    ref={compareLinkRef}
+                    href={`${diffSnapshotUrl}?from=${compareFrom}&to=${snapshotName}`}
+                    target="_blank"
+                    hidden
+                  ></a>
+                  <Icon
+                    className="clickable"
+                    name="random"
+                    onClick={() => {
+                      setIsOpen(false)
+                      compareLinkRef?.current?.click()
+                    }}
+                  />
+                </>
+              ) : null} */}
+            </div>
+          </Grid.Row>
+        ))}
+      </>
+    )
+  }
+
+  const renderLoadingAndError = () => (
+    <Portal open={isPortalOpen}>
+      <Segment
+        style={{
+          left: '40%',
+          position: 'fixed',
+          top: '50%',
+          minWidth: 100,
+          minHeight: 100,
+          zIndex: 1000,
+        }}
+      >
+        {isSnapshotError ? (
+          <Label size="large" color="red">
+            Error{' '}
+            <Icon
+              name="close"
+              onClick={() => {
+                setIsSnapshotError(false)
+                setIsPortalOpen(false)
+              }}
+            />
+          </Label>
+        ) : (
+          <Loader active size="small">
+            Loading
+          </Loader>
+        )}
+      </Segment>
+    </Portal>
+  )
+
+  const newSnapshot = () => {
+    const [value, setValue] = useState('')
+    if (compareFrom !== '') return null
+    return (
+      <Grid.Row key={`app_menu_new-snapshot`}>
+        <div>
+          <Input
+            size="mini"
+            onChange={(_, { value }) => setValue(value)}
+            placeholder="New Snapshot"
+          />
+          <Icon className="clickable" name="record" onClick={() => takeSnapshot(value)} />
+        </div>
+      </Grid.Row>
+    )
+  }
+
+  const randerUploadSnapshot = () => {
+    const fileInputRef = useRef<HTMLInputElement>(null)
+    if (compareFrom !== '') return null
+    // />
+    return (
+      <Grid.Row key={`app_menu_upload-snapshot`}>
+        <Button size="mini" onClick={() => fileInputRef?.current?.click()}>
+          Upload Snapshot {''}
+          <Icon name="upload" />
+        </Button>
+        <input
+          type="file"
+          ref={fileInputRef}
+          accept=".zip"
+          hidden
+          name="file"
+          multiple={false}
+          onChange={(e) => uploadSnapshot(e)}
+        />
+      </Grid.Row>
+    )
+  }
+
+  return (
+    <>
+      <Popup
+        position="bottom right"
+        trigger={<Icon name="angle down" style={{ paddingLeft: 10 }} />}
+        on="click"
+        onOpen={() => setIsOpen(true)}
+        onClose={() => setIsOpen(false)}
+        open={isOpen}
+        style={{ zIndex: 20 }}
+      >
+        <Grid textAlign="center" divided columns="equal">
+          <GridColumn>
+            {newSnapshot()}
+            {reanderSnapshotList()}
+            {randerUploadSnapshot()}
```

Comment:
Typos x2

Reply:
wow soo many typos, and even the branch name lol

552e1ef8c8a37bfa35e25b070eae62fd326b5df2